plot.py

```diff
@@ -0,0 +1,218 @@
+                mongo.get_collection(
+                    collection, {collection.get_meta_field_name(): subcategory}
+                )
+            )
+        df_for_subcategory = _densify(df_for_subcategory, collection, DENSIFY_UP_TO)
+        df_for_subcategory[collection.get_meta_field_name()] = subcategory
+        dfs_to_display.append(df_for_subcategory)
+    final_df = pd.concat(dfs_to_display)
+
+    daterange_start = date.today().strftime(DATE_FORMAT)
+    daterange_end = (date.today() - timedelta(days=DEFAULT_DATE_RANGE)).strftime(
+        DATE_FORMAT
+    )
+
+    return px.line(
+        final_df,
+        x="date",
+        y=collection.get_value_field_name(),
+        color=collection.get_meta_field_name(),
+    ).update_layout(xaxis_range=[daterange_end, daterange_start])
+
+
+def _transform(df: pd.DataFrame, collection: MongoCollection) -> pd.DataFrame:
+    match collection:
+        case MongoCollection.GoogleBounceRatePerDay:
+            df[collection.get_value_field_name()] = (
+                df.drop(["_id", "date"], axis=1).rolling(window=WINDOWS_SIZE).mean()
+            )
+    return df
+
+
+def _densify(
+    df: pd.DataFrame, collection: MongoCollection, how_many_days_back: int
+) -> pd.DataFrame:
+    match collection:
+        case MongoCollection.HexCumulativePackagesDownloads:
+            pass
+        case MongoCollection.DiscordMembersAtDay:
+            pass
+        case default:
+            df.set_index("date", drop=True, inplace=True)
+            df.index = pd.to_datetime(df.index)
+            idx = pd.date_range(
+                date.today() - timedelta(days=how_many_days_back), date.today()
+            )
+            df = df.reindex(idx)
+            df[df[collection.get_value_field_name()].isnull()] = 0
+            df["date"] = df.index
+            df = _transform(df, collection)
+    return df
+
+
+def _get_collections_for_metric_type(metric_type: str) -> MongoCollection:
+    match metric_type:
+        case "Discord":
+            return [
+                MongoCollection.DiscordMembersAtDay,
+                MongoCollection.DiscordMessagesPerDayPerChannel,
+            ]
+        case "Hex":
+            return [MongoCollection.HexCumulativePackagesDownloads]
+        case "Google Analytics":
+            return [
+                MongoCollection.GoogleBounceRatePerDay,
+                MongoCollection.GoogleTimeSpentPerDay,
+                MongoCollection.GoogleUsersFromTrafficSourcePerDay,
+                MongoCollection.GoogleUsersInTutorialPerDay,
+            ]
+
+
+def _get_default_subcategory(collection) -> list[str]:
+    match collection:
+        case MongoCollection.DiscordMessagesPerDayPerChannel:
+            return ["Total", "🙋help", "📋general"]
+        case MongoCollection.GoogleUsersFromTrafficSourcePerDay:
+            return ["Total", "(direct)", "google", "t.co", "github.com"]
+        case default:
+            return ["Total"]
```

Comment:
Indeed, it should be in the `collection.py`. When it comes to the second suggestion, I believe that changing of the emoji would generally break the system since in the database there still would be entries with the old emoji as the channel field :D We would need to provide additional logic to somehow "merge" the entries before the emoji change and the new entries, and since we probably do not plan to do it in predictable future, I will leave the defaults as they are

Reply:
Ok